src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/DynamicTable.cs

```diff
@@ -67,18 +69,15 @@ public void Insert(int? staticTableIndex, ReadOnlySpan<byte> name, ReadOnlySpan<
                 return;
             }
 
-            var entry = new HeaderField(staticTableIndex, name, value);
-            _buffer[_insertIndex] = entry;
-            _insertIndex = (_insertIndex + 1) % _buffer.Length;
-            _size += entry.Length;
-            _count++;
-        }
-
-        public void Resize(int maxSize)
-        {
-            if (maxSize > _maxSize)
+            // Ensure that we have at least one slot available.
+            if (_count == _buffer.Length)
             {
-                var newBuffer = new HeaderField[maxSize / HeaderField.RfcOverhead];
+                Debug.Assert(_count + 1 <= _maxSize / HeaderField.RfcOverhead);
```

Comment:
Consider putting `_maxSize / HeaderField.RfcOverhead` into a local (maxCount or whatever) so that it's clear we're validating _count against the same expression that's used in resizing.